development/webpack/utils/config.ts

```diff
@@ -78,6 +81,63 @@ export function getBuildName(
   return buildName;
 }
 
+/**
+ * Resolves the MetaMask build environment.
+ *
+ * The environment determines which Sentry project events are sent to,
+ * feature flag detection, and other build-specific behaviors.
+ *
+ * Resolution order:
+ * 1. If `--test` is set, returns 'testing'
+ * 2. If `--metamaskEnvironment` is explicitly set via CLI, uses that value
+ * 3. If `--env development`, returns 'development'
+ * 4. Otherwise, auto-detects from git context (release branch, main, PR, or other)
+ *
+ * NOTE: 'production' environment is NEVER auto-detected. It must be explicitly
+ * set via --metamaskEnvironment to prevent accidental pollution of production
+ * Sentry with events from local or CI test builds.
+ *
+ * @param args - The parsed CLI arguments
+ * @returns The resolved environment string
+ */
+export function resolveEnvironment(
```

Comment:
Previously, webpack just used `args.test ? 'testing' : env`, which meant local `--env production` builds would report as **production** to Sentry. Now:
- production is NEVER auto-detected, it must be explicitly set
- Git context is checked for proper environment detection